service/raft/group0_state_machine.cc

```diff
@@ -451,11 +490,22 @@ future<> group0_state_machine::transfer_snapshot(raft::server_id from_id, raft::
         co_await _sp.get_db().local().flush(db::system_keyspace::NAME, db::system_keyspace::TOPOLOGY);
     }
 
+    client_routes_service::client_route_keys client_routes_update;
     if (raft_snp) {
+        if (_sp.data_dictionary().has_schema(db::system_keyspace::NAME, db::system_keyspace::CLIENT_ROUTES)) {
+            auto s_client_routes = db::system_keyspace::client_routes();
+            for (auto& canonical_mut : raft_snp->mutations) {
+                if (canonical_mut.column_family_id() == s_client_routes->id()) {
+                    auto mut = co_await to_mutation_gently(canonical_mut, s_client_routes);
+                    collect_client_routes_update(mut, client_routes_update);
+                }
+            }
+        }
```

Comment:
We always send the event when `client_routes` is non-empty, so if the node that is applying the snapshot has the latest state of `client_routes`, the event will be send unnecessarily. However, snapshot transfer is rare, so this shouldn't be an issue. Also, I have no idea how to fix this. It looks like snapshot transfer isn't made to react on new changes.